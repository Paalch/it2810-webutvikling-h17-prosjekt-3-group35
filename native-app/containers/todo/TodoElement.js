import React from 'react';
import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
import PropTypes from 'prop-types';

export default class TodoElement extends React.Component {
    render() {
        return (
            <View>
                <input
                    style={styles.todoElement}
                    type={'checkbox'}
                    onProgress={() => this.props.removeElement(this.props.text)}/>
                <label>{this.props.text}</label>
            </View>
        );
    }
}

TodoElement.propTypes = {
    text: PropTypes.string.isRequired
};

const styles = StyleSheet.create({
    todoElement: {
        fontWeight: "normal",
    },
});

