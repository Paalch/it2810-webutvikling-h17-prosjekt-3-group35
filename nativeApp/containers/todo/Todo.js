import React from 'react';
import StyleSheet from 'react-native';
import Text from 'react-native';
import View from 'react-native';

class Todo extends React.Component {
  render () {
    return (
      <View Style={styles.todoContainer}>
        <View Style={styles.boxHeader}>
          <Text>2do</Text>
        </View>
        <View Style={styles.boxBody}>
          <Text>Lorem ipsum</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
 todoContainer: {
    backgroundColor: "#ccf",
    padding: 30,
    shadowColor: 'rgba(33,33,33,.7)',
     shadowOpacity: 7,
     shadowOffset: {
         height: 5,
         width: 5
     },
 },

 boxheader: {
     flexDirection: 'row'
 },

 boxBocy: {
     flexDirection: 'row'
 }

});

export default Todo;