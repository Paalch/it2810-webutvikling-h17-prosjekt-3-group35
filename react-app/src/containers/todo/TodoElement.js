import React, {Component} from 'react';


class TodoElement extends Component {

  render() {
    return (
      <div className={'todo-element'}>
        <input
          className={'checkbox-round'}
          type={'checkbox'}
          checked={''}
          onClick={() => this.props.removeElement(this.props.todo.id)}/>
        <label>{this.props.todo.value}</label>
      </div>
    )
  }
}

TodoElement.propTypes = {

};

export default TodoElement;