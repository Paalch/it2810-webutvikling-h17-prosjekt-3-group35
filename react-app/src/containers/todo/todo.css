.todo-container {
  background: #212a33;
  padding: 20px;
  box-shadow: 5px 5px 7px rgba(33,33,33,.7);
  opacity: 1;
  max-height: 40vh;
  overflow: scroll;
}

.todo-header {
  margin-bottom: 3%;
  padding-left: 20px;
  padding-top: 10px;
  background-color: mediumpurple;
  margin-top: -20px;
  margin-left: -20px;
  margin-right: -20px;

}

.todo-body {
  background-color: #ccccff;
  padding: 20px;
}

.todo-add-button {
  background-color: #ccccff;
  font-size: 2em;
  padding: 0px;
  border: none;
}

.todo-add-button:focus, .todo-add-button:active, .todo-add-button:hover{
  background-color: #ccccff !important;
  border: none !important;
  outline: none !important;
  box-shadow: none !important;
}

.form-control:focus {
  outline: none;
}

#todo-form-input {
  background: #ccf;
  border: none;
  border-radius: 0;
  box-shadow: none;
}

#todos {
  margin-top: 10px;
}

.todo-element label {
  font-weight: normal;
}

.checkbox-round {
  margin: 5px 7px 5px 5px !important;

  width: 1.3em;
  height: 1.3em;
  background-color: white;
  border-radius: 50%;
  vertical-align: middle;
  border: 1px solid #ddd;
  -webkit-appearance: none;
  outline: none;
  cursor: pointer;
}

.checkbox-round:checked {
  background-color: #777777;
}

.checkbox-round:active, .checkbox-round:focus {
  outline: none !important;
}