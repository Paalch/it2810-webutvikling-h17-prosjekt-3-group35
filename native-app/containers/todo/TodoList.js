import React from 'react';
import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
import TodoElement from './TodoElement';

export default class TodoList extends React.Component {
    constructor() {
        super();
        this.state = {
            todos: [],
            text: ''
        };
        this.updateTextField = this.updateTextField.bind(this);
        this.saveTodo = this.saveTodo.bind(this);
        this.removeElement = this.removeElement.bind(this);
    }

    updateTextField(e) {
        this.setState({text: e.target.value});
    }

    removeElement(todo) {
        this.setState({
            todos: this.state.todos.filter((t) =>
                t !== todo
            )
        });
    }

    saveTodo() {
        const todoValue = this.state.text;
        if (todoValue.length === 0) {
            return;
        }
        this.setState({todos: this.state.todos.concat(todoValue), text: ''});
    };

    render() {
        const content = this.state.todos.map((todo, i) =>
            <TodoElement text={todo} key={i} removeElement={this.removeElement}/>
        );
        return (
            <View style={styles.container}>
                <View>
                    <View>
                        <TextInput
                            style={styles.formInput}
                            type={'text'}
                            placeholder={'.............................................................'}
                            value={this.state.text}
                            onChange={this.updateTextField}
                        />
                    </View>
                    <View>
                        <Button onPress={this.saveTodo} title={'+'}/>
                    </View>
                </View>
                <View style={styles.todos}>
                    {content}
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#ccccff",
        padding: 20,
    },

    todos: {
        marginTop: 10,
    },

});
