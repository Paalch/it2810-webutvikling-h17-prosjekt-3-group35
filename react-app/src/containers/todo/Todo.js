import React from 'react';
import './todo.css';

class Todo extends React.Component {
  render () {
    return (
      <div className={'todo-container box'}>
        <h1>Todo</h1>
      </div>
    )
  }
}

export default Todo;