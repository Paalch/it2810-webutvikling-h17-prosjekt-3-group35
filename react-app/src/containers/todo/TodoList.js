import React, {Component} from 'react';
import {Col, Row, FormControl, Button, FormGroup, InputGroup} from 'react-bootstrap';
import TodoElement from "./TodoElement";
import './todo.css';

class TodoList extends Component {

  constructor() {
    super();
    this.state = {
      todos: [],
      text: ''
    };
    this.updateTextField = this.updateTextField.bind(this);
    this.saveTodo = this.saveTodo.bind(this);
    this.removeElement = this.removeElement.bind(this);
    this.save = this.save.bind(this);
    this.load = this.load.bind(this);
  }

  componentDidMount() {
    const data = this.load();
    if (data !== null) {
      this.setState(() => data);
    }
  }

  componentDidUpdate() {
    this.save();
  }

  save() {
    localStorage.setItem('todo', JSON.stringify(this.state));
  }

  load() {
    return 'todo' in localStorage ?
      JSON.parse(localStorage.getItem('todo')) : null;
  }

  updateTextField(e) {
    this.setState({text: e.target.value});
  }

  removeElement(id) {
    this.setState({
      todos: this.state.todos.filter((t) =>
        id !== t.id
      )
    });
  }

  saveTodo() {
    const todoValue = this.state.text;
    if (todoValue.length === 0) {
      return;
    }
    const newTodo = {id: Date.now(), value: todoValue};
    this.setState({todos: this.state.todos.concat(newTodo), text: ''});
  };

  render() {
    const content = this.state.todos.map((element, i) =>
      <TodoElement todo={element} key={i} removeElement={this.removeElement}/>
    );

    return (
      <div>
        <Row>
          <Col xs={10}>
            <FormControl
              id={'todo-form-input'}
              type={'text'}
              placeholder={'.............................................................'}
              value={this.state.text}
              onChange={this.updateTextField}
            />
          </Col>
          <Col xs={2}>
            <Button
              className={'todo-add-button'}
              onClick={this.saveTodo}>
              +
            </Button>
          </Col>
        </Row>
        <div id={'todos'}>
          {content}
        </div>
      </div>
    )
  }
}

export default TodoList;