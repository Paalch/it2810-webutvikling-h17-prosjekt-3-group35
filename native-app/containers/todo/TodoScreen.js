import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import TodoList from './TodoList';

export default class TodoScreen extends React.Component {
    static navigationOptions = {
        title: 'Todo',
        headerStyle: { backgroundColor: '#212a34', elevation: 6, paddingTop: 6,},
        headerTitleStyle: { color: '#fff',},
        headerTintColor: '#fff'
    };
    render() {
        return (
            <View style={styles.container}>
                <View style={styles.todoBody}>
                    <TodoList />
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#666",
        padding: 10,
    },
});

