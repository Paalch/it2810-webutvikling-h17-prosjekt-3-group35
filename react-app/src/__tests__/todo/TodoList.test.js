import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import TodoList from '../../containers/todo/TodoList'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
import { saveTodo } from '../../containers/todo/TodoList'
Enzyme.configure({ adapter: new Adapter() })
import { Button } from 'react-bootstrap';


describe('Checks that the todo components renders correctly', () => {

    it('renders correctly', () => {
        const rendered = renderer.create(
            <TodoList />
        )
        expect(rendered.toJSON()).toMatchSnapshot()
    })

    it('Test that a empty value can not be added', () => {
        const wrapper = shallow(<TodoList/>)
        wrapper.find('Button').simulate('click');
        let state = wrapper.state().todos
        expect(state).toHaveLength(0)
    })

    it('Test that you can create a new todo', () => {
        const wrapper = shallow(<TodoList />)
        const input = wrapper.find('FormControl')
        input.value = 'I am a new todo'
        //Since javascript is async i need to wait for the state to update before i check the new state
        setTimeout(function(){
            wrapper.find('Button').simulate('click')
            let state = wrapper.state().todos
            expect(state).toHaveLength(1)
        }, 1);
    })

})