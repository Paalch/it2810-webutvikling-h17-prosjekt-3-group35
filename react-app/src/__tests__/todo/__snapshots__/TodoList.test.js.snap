// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the todo components renders correctly renders correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xs-10"
    >
      <input
        className="form-control"
        id="todo-form-input"
        onChange={[Function]}
        placeholder="............................................................."
        type="text"
        value=""
      />
    </div>
    <div
      className="col-xs-2"
    >
      <button
        className="todo-add-button btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      />
    </div>
  </div>
  <div
    id="todos"
  />
</div>
`;
